src/AzureAppConfigurationImpl.ts

```diff
@@ -196,17 +240,65 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
             keyValues.push([key, value]);
         }
 
-        this.#configMap.clear(); // clear existing key-values in case of configuration setting deletion
+        this.#clearLoadedKeyValues(); // clear existing key-values in case of configuration setting deletion
         for (const [k, v] of keyValues) {
             this.#configMap.set(k, v);
         }
     }
 
+    async #clearLoadedKeyValues() {
+        for (const key of this.#configMap.keys()) {
+            if (key !== FEATURE_MANAGEMENT_KEY_NAME) {
+                this.#configMap.delete(key);
+            }
+        }
+    }
+
+    async #loadFeatureFlags() {
+        // Temporary map to store feature flags, key is the key of the setting, value is the raw value of the setting
+        const featureFlagsMap = new Map<string, any>();
+        for (const selector of this.#featureFlagSelectors) {
+            const listOptions: ListConfigurationSettingsOptions = {
+                keyFilter: `${featureFlagPrefix}${selector.keyFilter}`,
+                labelFilter: selector.labelFilter
+            };
+
+            const pageEtags: string[] = [];
+            const pageIterator = listConfigurationSettingsWithTrace(
+                this.#requestTraceOptions,
+                this.#client,
+                listOptions
+            ).byPage();
+            for await (const page of pageIterator) {
+                if (page._response.status === 200) {
+                    if (page.etag) {
+                        pageEtags.push(page.etag);
+                    }
+                }
+                for (const setting of page.items) {
+                    if (isFeatureFlag(setting)) {
+                        featureFlagsMap.set(setting.key, setting.value);
+                    }
+                }
+            }
+            selector.pageEtags = pageEtags;
+        }
+
+        // parse feature flags
+        const featureFlags = Array.from(featureFlagsMap.values()).map(rawFlag => JSON.parse(rawFlag));

```

Comment:
What's the behavior if there are failures while parsing FF due to invalid JSON?

Reply:
It would throw a SyntaxError (by JSON.parse), same as SDK [does](https://github.com/Azure/azure-sdk-for-js/blob/d328d966b33449e58b729ef1cc792fe307b64ff4/sdk/appconfiguration/app-configuration/src/featureFlag.ts#L102). And it fails the whole FF loading.

Invalid JSON in FF is a user error that we expect them to fix it first by themselves. I think it makes sense to simply throw the error and break the load.